packages/hdwallet-core/src/utils.ts

```diff
@@ -208,3 +208,7 @@ export function compatibleBufferConcat(list: Uint8Array[]): Buffer {
   if (!checkBufferConcat()) return Buffer.concat(list);
   return Buffer.concat(list.map(x => Buffer.isBuffer(x) ? x : Buffer.from(x)));
 }
+
+export function isIndexable(x: unknown): x is Record<string, unknown> {
+  return x !== null && ["object", "function"].includes(typeof x)
```

Comment:
save yourself a few CPU cycles and check undefined up-front?
```suggestion
  return x != null && ["object", "function"].includes(typeof x)
```

Reply:
I'd argue that comparing with to `null` or `undefined` with `!=` is dangerous in general and the benefits of avoiding having to think about why this instance is OK outweigh any performance impact.